export_dagmc_cmd/mcnp2cad.cpp

```diff
@@ -0,0 +1,1325 @@
+//  CHK_MB_ERR_RET("Error parsing options: ",rval);
+
+
+  try{
+    std::ifstream input( filename.c_str(), std::ios::in );
+    InputDeck& deck = InputDeck::build(input);
+    record << "filename is " << filename << std::endl;
+  
+    // turn off debug if it was set by debug_input only
+    if( din ){ Gopt.debug = false; }
+    
+    if( dout && !OPT_DEBUG ){ Gopt.debug = true; }
+  
+    if( OPT_DEBUG ){ 
+      debugSurfaceDistances( deck );
+    }
+  
+    iGeom_Instance igm;
+    int igm_result;
+  
+    iGeom_newGeom( Gopt.igeom_init_options.c_str(), &igm, &igm_result, Gopt.igeom_init_options.length() );
+    CHECK_IGEOM( igm_result, "Initializing iGeom" );
+  
+    GeometryContext context( igm, deck );
+    context.createGeometry();
+  }
+  catch(std::runtime_error& e){
+    record << e.what() << std::endl;
+    PRINT_INFO( e.what() );
+    PRINT_INFO( "\n" );
+    std::cerr << e.what() << std::endl;
+    record.close();
+    return false;
+  }
+  
+  record.close();
+  return true;
+}
+
+bool MCNP2CAD::parse_options(CubitCommandData &data)
+{
+
+  // read parsed command for tolerance
+  data.get_value("specific_tolerance",Gopt.specific_tolerance);
+  record << "Setting specific tolerance to " << Gopt.specific_tolerance << std::endl;
+  //If tolerance was specified
+  if( data.find_keyword("tol") ){
+    Gopt.override_tolerance = true;
+    if( Gopt.specific_tolerance <= 0.0 || Gopt.specific_tolerance > .1 ){
+      std::cerr << "Warning: you seem to have specified an unusual tolerance (" 
+                << Gopt.specific_tolerance << ")." << std::endl;
+      record << "Warning: you seem to have specified an unusual tolerance (" 
+             << Gopt.specific_tolerance << ")." << std::endl;
+    }
+  }
+  
+  // read parsed boolean commands
+  Gopt.verbose = data.find_keyword("verbose");
+  Gopt.debug = data.find_keyword("debug");
+  Gopt.infinite_lattice_extra_effort = data.find_keyword("extra_effort");
+  Gopt.tag_materials = !data.find_keyword("skip_mats");
+  Gopt.tag_importances = !data.find_keyword("skip_imps");
+  Gopt.tag_cell_IDs = !data.find_keyword("skip_nums");
+  Gopt.make_graveyard = !data.find_keyword("skip_graveyard");
+  Gopt.merge_geom = !data.find_keyword("skip_merge");
+  Gopt.imprint_geom = !data.find_keyword("skip_imprint");
+  Gopt.uwuw_names = data.find_keyword("uwuw_names");
+  dout = data.find_keyword("debug_output");
+  din  = data.find_keyword("debug_input");
+
+  if( Gopt.merge_geom && !Gopt.imprint_geom ) {
+    record << "Warning: cannot merge geometry without imprinting, will skip merge too." << std::endl;
+    std::cerr << "Warning: cannot merge geometry without imprinting, will skip merge too." << std::endl;
+  }
+  
+  record << "Reading input file..." << std::endl;
+
+  // if debug_input and not debug, set debugging to be true for InputDeck::build() call only
+  
+  if( din && !OPT_DEBUG ){
+    Gopt.debug = true;
+  }
+  else{ din = false; }
```

Comment:
Need consistency with writing a statement on one line (the else statement) or multi-line (the if part). The if part here could probably be written on one line like the else part.